lib/exchanges.js

```diff
@@ -0,0 +1,94 @@
+var base      = require('taskcluster-base');
+var assert    = require('assert');
+var _         = require('lodash');
+
+// Common schema prefix
+var SCHEMA_PREFIX_CONST = 'http://schemas.taskcluster.net/github/v1/';
+
+/** Build common routing key construct for `exchanges.declare` */
+var commonRoutingKey = function(options) {
+    options = options || {};
+     let routingKey = [
+        {
+          name:             'routingKeyKind',
+          summary:          "Identifier for the routing-key kind. This is " +
+                            "always `'primary'` for the formalized routing key.",
+          constant:         'primary',
+          required:         true
+        }, {
+          name:             'organization',
+          summary:          "The GitHub `organization` which had an event.",
+          maxSize:          100,
+          required:         true
+        }, {
+          name:             'repository',
+          summary:          "The GitHub `repository` which had an event.",
```

Comment:
We have a problem here. Github repository names may container `[A-Za-z0-9_.-]` possibly more, I wasn't able to find documentation...

Anyways, routing key entries cannot contain `.` as that is the separator. Since, github doesn't allow `%` perhaps we should write that dots are replaced with `%` or maybe it's more sane to replace dots with `-dot-`, it's not like it's going to create a lot of conflicts... Or `%dot%` maybe?

Eitherway, we can't have the dots there... Same goes for organization, I'm not sure what the character limits for github actually are, we should look a bit into that...